enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyUtils.java

```diff
@@ -0,0 +1,157 @@
+import org.apache.logging.log4j.Logger;
+import org.wso2.choreo.connect.enforcer.config.ConfigHolder;
+import org.wso2.choreo.connect.enforcer.util.FilterUtils;
+
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.text.ParseException;
+import java.util.Base64;
+import java.util.Date;
+import java.util.Optional;
+import java.util.zip.CRC32;
+
+/**
+ * This class holds the utility methods related to API key functionalities.
+ */
+public class APIKeyUtils {
+
+    private static final Logger log = LogManager.getLogger(APIKeyUtils.class);
+
+    /**
+     * Check if the provided API key is valid.
+     *
+     * @param apiKey    API Key
+     * @return          True if valid.
+     */
+    public static boolean isValidAPIKey(String apiKey) {
+
+        // Getting the API Key checksum.
+        String checksum = apiKey.substring(apiKey.length() - 6);
+        // Skipping the prefix(`chp_`) and checksum.
+        String keyData = apiKey.substring(4, apiKey.length() - 6);
+        String generatedChecksum = generateCRC32Checksum(keyData);
+        return StringUtils.equals(checksum, generatedChecksum);
+    }
+
+    /**
+     * Generate API Key Hash.
+     *
+     * @param apiKey    API Key
+     * @return key hash
+     */
+    public static String generateAPIKeyHash(String apiKey) {
+
+        // Skipping the prefix(`chp_`) and checksum.
+        String keyData = apiKey.substring(4, apiKey.length() - 6);
+        // Base 64 decode key data.
+        String decodedKeyData = new String(Base64.getDecoder().decode(keyData));
+        // Convert data into JSON.
+        JSONObject jsonObject = (JSONObject) JSONValue.parse(decodedKeyData);
+        // Extracting the key.
+        String key = jsonObject.getAsString(APIKeyConstants.API_KEY_JSON_KEY);
+        // Return SHA256 hash of the key.
+        return DigestUtils.sha256Hex(key);
+    }
+
+    /**
+     * This function exchanges a given API key to an JWT token.
+     *
+     * @param keyHash    Key hash
+     * @return          JWT corresponding to given key hash.
+     */
+    public static Optional<String> exchangeAPIKeyToJWT(String keyHash) {
+
+        URL url = null;
+        try {
+            url = new URL(ConfigHolder.getInstance().getConfig().getApiKeyConfig().getOauthAgentURL());
+        } catch (MalformedURLException e) {
+            log.error("Error occurred while parsing OAuth agent URL", e);
+            return Optional.empty();
+        }
+        try (CloseableHttpClient httpClient = (CloseableHttpClient) FilterUtils.getHttpClient(url.getProtocol())) {
+            // Create a request to exchange API key to JWT.
+            HttpPost exchangeRequest = new HttpPost(url.toURI());
+            exchangeRequest.addHeader("Content-Type", ContentType.APPLICATION_JSON.toString());
+            exchangeRequest.setEntity(new StringEntity("{\"apiKeyHash\": \"" + keyHash + "\"}"));
```

Comment:
Move these string to constant

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/3592/commits/8e7a16edf9d71b7676631480e5824113ff5b30de